config/300-kafkasource.yaml

```
@@ -0,0 +1,399 @@
+                    type: string
+                    enum: [GSSAPI, PLAIN, SCRAM-SHA-256, SCRAM-SHA-512]
+                  username:
+                    description: Kafka account username.
+                    type: string
+                  password:
+                    description: Kafka account password.
+                    type: object
+                    properties:
+                      value:
+                        description: Plain text password.
+                        type: string
+                      valueFromSecret:
+                        description: A reference to a Kubernetes Secret object containing the password.
+                        type: object
+                        properties:
+                          name:
+                            type: string
+                          key:
+                            type: string
+                        required:
+                        - name
+                        - key
+                    oneOf:
+                    - required: [value]
+                    - required: [valueFromSecret]
+                  kerberosAuth:
+                    description: Kerberos Authentication method to interact with Kafka.
+                    type: object
+                    properties:
+                      kerberosRealm:
+                        description: The name of the Kerberos Realm.
+                        type: string
+                      kerberosServiceName:
+                        description: The primary name of the Kafka server configured.
+                        type: string
+                      username:
+                        description: Kerberos Username.
+                        type: string
+                      password:
+                        description: Kerberos Password.
+                        type: object
+                        properties:
+                          value:
+                            description: Plain text password.
+                            type: string
+                          valueFromSecret:
+                            description: A reference to a Kubernetes Secret object containing the password.
+                            type: object
+                            properties:
+                              name:
+                                type: string
+                              key:
+                                type: string
+                            required:
+                            - name
+                            - key
+                        oneOf:
+                        - required: [value]
+                        - required: [valueFromSecret]
+                      kerberosConfig:
+                        type: object
+                        properties:
+                          valueFromSecret:
+                            description: A reference to a Kubernetes Secret object containing the value.
```

Descriptions is what our UI use to help people understand it, so maybe ...
```suggestion
                            description: A reference to a Kubernetes Secret object containing the kerberos configuration file (krb5.conf).
```